bpf/tests/tc_nodeport_lb4_test.c

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+#include "common.h"
+
+#include "bpf/builtins.h"
+#include "bpf/ctx/skb.h"
+
+#define ENABLE_IPV4
+#define ENABLE_NODEPORT
+
+#define fib_lookup mock_fib_lookup
+
+static const char fib_smac[6] = {0xDE, 0xAD, 0xBE, 0xEF, 0x01, 0x02};
+static const char fib_dmac[6] = {0x13, 0x37, 0x13, 0x37, 0x13, 0x37};
+
+long mock_fib_lookup(__maybe_unused void *ctx, struct bpf_fib_lookup *params,
+		     __maybe_unused int plen, __maybe_unused __u32 flags)
+{
+	memcpy(params->smac, fib_smac, sizeof(fib_smac));
+	memcpy(params->dmac, fib_dmac, sizeof(fib_dmac));
+	return 0;
+}
+
+#include "config_replacement.h"
+
+#include "bpf_host.c"
+#include "lib/nodeport.h"
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(max_entries, 2);
+	__array(values, int());
+} entry_call_map __section(".maps") = {
+	.values = {
+		[0] = &from_netdev,
+	},
+};
+
+#define BACKEND_IP 0x0F00020A /* 10.2.0.15 */
+#define BACKEND_PORT 8080
+
+static long (*bpf_skb_change_tail)(struct __sk_buff *skb, __u32 len, __u64 flags) = (void *)38;
+
+static inline int ctx_recalc(struct __ctx_buff *ctx, int offset)
+{
+	return ctx_full_len(ctx) + offset;
+}
+
+SETUP("tc", "forward_to_other_node")
```

Comment:
Perhaps we want to give this a different name to not clash with the existing `forward_to_other_node` test.